yt/frontends/amrex/data_structures.py

```diff
@@ -652,7 +652,7 @@ def __init__(
 
         cparam_filename = cparam_filename or self.__class__._default_cparam_filename
         self.cparam_filename = self._lookup_cparam_filepath(
-            output_dir, cparam_filename=cparam_filename
+            self.output_dir, cparam_filename=cparam_filename
```

Comment:
It should be possible to use self.directory instead provided the parent class' `__init__` is called first. Can you try that ?

Reply:
Unfortunately that doesn't work: `Dataset.__init__()` calls `self._parse_parameter_file()`, which needs `self.cparam_filename` to be set beforehand.

Reply:
fair enough !